sdk/storage/azure_storage_blob/tests/blob_client.rs

```diff
@@ -15,348 +19,426 @@ use std::error::Error;
     let blob_name = recording
         .random_string::<12>(Some("blob"))
         .to_ascii_lowercase();
 
+    let container_client_options = BlobContainerClientOptions {
+        client_options: options.clone(),
+        ..Default::default()
+    };
     // Act
-    let container_client = ContainerClient::new(
+    let container_client = BlobContainerClient::new(
         &endpoint,
         container_name.clone(),
         recording.credential(),
-        Some(options.clone()),
+        Some(container_client_options),
     )?;
-    container_client.create_container(None).await?;
 
+    let blob_client_options = BlobClientOptions {
+        client_options: options.clone(),
+        ..Default::default()
+    };
     let blob_client = BlobClient::new(
         &endpoint,
         container_name,
         blob_name,
         recording.credential(),
-        Some(options),
+        Some(blob_client_options),
     )?;
+
+    // Invalid Container Scenario
+    let response = blob_client.get_blob_properties(None).await;
+
+    // Assert
+    let error = response.unwrap_err().http_status();
+    assert_eq!(StatusCode::NotFound, error.unwrap());
+
+    container_client.create_container(None).await?;
+
     let data = b"hello rusty world";
+
     blob_client
         .upload_blob(
             RequestContent::from(data.to_vec()),
             true,
-            i64::try_from(data.len())?,
+            u64::try_from(data.len())?,
             None,
         )
         .await?;
-    let response = blob_client.get_blob_properties(None).await;
-
-    // Assert
-    assert!(response.is_ok());
 
-    let blob_properties = response?;
-    assert_eq!(blob_properties.blob_type, Some(BlobType::BlockBlob));
-    assert_eq!(blob_properties.content_length, Some(17));
-
-    container_client.delete_container(None).await?;
-    Ok(())
-}
-
-#[recorded::test]
-async fn test_get_blob_properties_invalid_container(
-    ctx: TestContext,
-) -> Result<(), Box<dyn Error>> {
-    // Recording Setup
-    let recording = ctx.recording();
-    let (options, endpoint) = recorded_test_setup(recording, BlobClientOptions::default()).await;
-    let container_name = recording
-        .random_string::<17>(Some("container"))
-        .to_ascii_lowercase();
-    let blob_name = recording
-        .random_string::<12>(Some("blob"))
-        .to_ascii_lowercase();
-
-    // Act
-    let blob_client = BlobClient::new(
-        &endpoint,
-        container_name,
-        blob_name,
-        recording.credential(),
-        Some(options),
-    )?;
-    let response = blob_client.get_blob_properties(None).await;
+    // No Option Scenario
+    let response = blob_client.get_blob_properties(None).await?;
 
     // Assert
-    assert_eq!(
-        String::from("HttpResponse(NotFound, \"ContainerNotFound\")"),
-        response.unwrap_err().kind().to_string()
-    );
+    let lease_state = response.lease_state()?;
+    assert_eq!(LeaseState::Available, lease_state.unwrap());
```

Comment:
Can we assert a couple more and more useful properties other than lease state? Maybe size and then maybe some ones that need to get serialized like Etag and a date? (obviously can't assert exact values there but just check they seem correct somehow)